tools/tu_collector/tu_collector/tu_collector.py

```diff
@@ -134,6 +134,8 @@ def __eliminate_argument(arg_vect, opt_string, has_arg=False):
         output, rc = oerr.strerror, oerr.errno
 
     output = codecs.decode(output, 'utf-8', 'replace')
+    # Remove the potential stderr messages from the beginning of the stdout.
+    output = output[output.find('__dummy'):]
```

Comment:
It would be better to redirect the standard error of the above command to `/dev/null` so these errors will not be the part of the output.

For example your solution will not solve the following problem:
```sh
gcc ./main.o -std=c++11 -E -M -MT __dummy main.c
```

Output:
```sh
cc1: warning: command line option ‘-std=c++11’ is valid for C++/ObjC++ but not for C
__dummy: ./main.c /usr/include/stdc-predef.h \
 /usr/include/stdio.h \
 /usr/include/x86_64-linux-gnu/bits/libc-header-start.h \
...
gcc: warning: ./main.o: linker input file unused because linking not done
```
The problem with this that there is an error message at the end of the output too.